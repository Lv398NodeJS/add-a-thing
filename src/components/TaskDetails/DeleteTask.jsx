import React from 'react';
import { Modal, Button } from 'react-bootstrap';

export default class DeleteTask extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showDelete: props.showDelete,
    };
  }

  handleDelete = () => {
    const { taskRef, closeTaskDetails } = this.props;
    const { showDelete } = this.state;
    taskRef.remove();
    this.setState({
      showDelete: !showDelete,
    });
    closeTaskDetails();
  };

  closeModal = () => {
    const { closeDeleteModal } = this.props;
    closeDeleteModal();
  };

  render() {
    const { showDelete } = this.props;
    return (
      <Modal
        show={showDelete}
        onHide={this.closeModal}
        aria-labelledby="contained-modal-title-center"
        centered
      >
        <Modal.Header>
          <Modal.Title>Delete task</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <p>Are you sure ?</p>
        </Modal.Body>
        <Modal.Footer>
          <Button
            variant="outline-primary"
            onClick={this.closeModal}
          >
            {'Cancel'}
          </Button>
          <Button
            variant="outline-danger"
            onClick={this.handleDelete}
          >
            {'Delete'}
          </Button>
        </Modal.Footer>
      </Modal>
    );
  }
}
