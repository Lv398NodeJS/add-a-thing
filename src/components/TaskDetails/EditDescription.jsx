import React from 'react';
import { Button, Container, Form } from 'react-bootstrap';

export default class EditDescription extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      editDescription: props.editDescription,
    };
  }

  handleSaveDescription = () => {
    const { closeEditDescriptionField, saveDescription } = this.props;
    const { editDescription } = this.state;
    saveDescription(false, this.taskDescription.value, false, false);
    closeEditDescriptionField();
    this.setState({ editDescription: !editDescription });
  };

  render() {
    const { description } = this.props;
    return (
      <Container>
        <Form.Label htmlFor="taskDescription">Description: </Form.Label>
        <Form.Control
          name="taskDescription"
          as="textarea"
          placeholder="Description"
          defaultValue={description}
          ref={(taskDescription) => {
            this.taskDescription = taskDescription;
          }}
        />
        <Button onClick={this.handleSaveDescription}>
          {'Save description'}
        </Button>
      </Container>
    );
  }
}
