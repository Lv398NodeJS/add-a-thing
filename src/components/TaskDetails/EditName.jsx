import React from 'react';
import { Container, Form, Button } from 'react-bootstrap';

export default class EditName extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      editName: props.editName,
    };
  }

  handleSaveName = () => {
    const { saveName, closeEditNameField } = this.props;
    const { editName } = this.state;
    if (this.taskName.value.trim() === '') return;
    saveName(this.taskName.value, false, false, false);
    closeEditNameField();
    this.setState({ editName: !editName });
  };

  closeName = () => {
    const { closeEditNameField } = this.props;
    closeEditNameField();
  };

  render() {
    const { name } = this.props;
    return (
      <Container>
        <Form.Label htmlFor="taskName">Name: </Form.Label>
        <Form.Control
          name="taskName"
          type="text"
          placeholder="Name"
          defaultValue={name}
          maxLength={30}
          ref={(taskName) => {
            this.taskName = taskName;
          }}
        />
        <Button
          className="button-save-task-details"
          onClick={this.handleSaveName}
        >
          {'Save name'}
        </Button>
        <Button
          className="cancel-button"
          onClick={this.closeName}
        >
          {'â•³'}
        </Button>
      </Container>
    );
  }
}
