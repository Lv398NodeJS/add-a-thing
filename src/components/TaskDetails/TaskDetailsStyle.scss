.modal{
  padding: 0;
}
.main-modal-for-taskN{
  font-family: sans-serif;
  font-size: 1.3rem;
  color: rgb(103, 102, 106);
}
.main-modal-for-taskD{
  font-family: sans-serif;
  font-size: 1rem;
  color: rgb(103, 102, 106);
}
.delete-task-details-button{
  max-width: 100px;
  margin: 5px 5px 5px 77%;
}
.button-save-task-details{
  margin: 5px;
}
.task-details-container{
  padding: 10px;
  min-height: 30px;
  margin: 10px;
  background-color: rgba(108, 117, 125, 0.07);
  border-radius: 3px;
}
.task-details-container{
  &:hover {
    cursor: pointer;
    background-color: rgba(108, 117, 125, 0.22);
  }
}
.description-container{
  min-height: 60px;
}

.dropdown-priority-container{
  color: rgba(0, 0, 0, 0.72);
}
.dropdown-container{
  margin: 2px;
}

.delete-modal-window{
  background-color: rgba(1, 1, 1, 0.6);
}
.High {
  &, &:hover, &:link, &:focus, &:active {
    background-color: red !important;
    }
  }

.Medium {
  &, &:hover, &:link, &:focus, &:active {
    background-color: orange !important;
  }
}
.Low {
  &, &:hover, &:link, &:focus, &:active {
    background-color: green !important;
  }
}

.cancel-button {
  &, &:hover, &:link, &:focus, &:active {
    background: none !important;
    border: none;
    outline: none;
    margin: -10px 0;
    box-shadow: none !important;
  }
}
.close-task-button {
  &:hover, &:link, &:focus, &:active {
    color: black;
  }
}

