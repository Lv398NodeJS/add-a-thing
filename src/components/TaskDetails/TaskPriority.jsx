import React from 'react';
import { Container, Dropdown } from 'react-bootstrap';

export default class TaskPriority extends React.Component {
  handleChangePriority = (priorityName) => {
    const { changePriority } = this.props;
    changePriority(false, false, false, priorityName);
  };

  render() {
    const { priority } = this.props;
    switch (priority) {
      case 'High':
        this.priorityColor = '#ff666b';
        break;

      case 'Medium':
        this.priorityColor = '#9effbd';
        break;

      case 'Low':
        this.priorityColor = '#fff98b';
        break;

      default:
        this.priorityColor = '#9fa1a3';
        break;
    }
    return (
      <Container>
        <Dropdown>
          <Dropdown.Toggle className="dropdownPriorityContainer col" size="sm" id="dropdown-basic" style={{ backgroundColor: this.priorityColor }}>
            {priority || 'Choose priority'}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <Dropdown.Item onClick={() => { this.handleChangePriority('High'); }}>
              {'High'}
            </Dropdown.Item>
            <Dropdown.Item onClick={() => { this.handleChangePriority('Medium'); }}>
              {'Medium'}
            </Dropdown.Item>
            <Dropdown.Item onClick={() => { this.handleChangePriority('Low'); }}>
              {'Low'}
            </Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Container>
    );
  }
}
