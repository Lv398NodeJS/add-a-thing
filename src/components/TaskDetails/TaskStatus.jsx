import React from 'react';
import { Container, Dropdown } from 'react-bootstrap';

export default class TaskStatus extends React.Component {
  handleChangeStatus = (statusName) => {
    const { changeStatus } = this.props;
    changeStatus(false, false, statusName, false);
  };

  render() {
    const { status } = this.props;
    return (
      <Container>
        <Dropdown>
          <Dropdown.Toggle className="dropdownContainer col" size="sm" variant="primary" id="dropdown-basic">
            {status}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <Dropdown.Item onClick={() => { this.handleChangeStatus('To Do'); }}>
              {'To Do'}
            </Dropdown.Item>
            <Dropdown.Item onClick={() => { this.handleChangeStatus('In Progress'); }}>
              {'In Progress'}
            </Dropdown.Item>
            <Dropdown.Item onClick={() => { this.handleChangeStatus('Done'); }}>
              {'Done'}
            </Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Container>
    );
  }
}
