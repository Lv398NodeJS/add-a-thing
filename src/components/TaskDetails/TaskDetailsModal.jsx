import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import TaskDetails from './TaskDetails';
import SubTaskList from '../SubTaskList/SubTaskList';
import DeleteTask from './DeleteTask';

export default class TaskDetailsModal extends React.Component {
  constructor() {
    super();
    this.state = {
      showDelete: false,
    };
  }

  closeTaskDetails = () => {
    const { onClose: close } = this.props;
    close();
  };

  closeDeleteModal = () => {
    const { showDelete } = this.state;
    this.setState({
      showDelete: !showDelete,
    });
  };

  render() {
    const { taskRef, show: modalShow } = this.props;
    const { showDelete } = this.state;
    return (
      <Modal
        show={modalShow}
        onHide={this.closeTaskDetails}
        aria-labelledby="contained-modal-title-center"
        centered
      >
        <Modal.Body>
          <TaskDetails
            taskRef={taskRef}
            closeTaskDetails={this.closeTaskDetails}
          />
          <SubTaskList taskRef={taskRef} />
        </Modal.Body>
        <Button
          variant="outline-danger"
          onClick={() => this.setState({ showDelete: !showDelete })}
          className="delete-task-details-button float-right"
        >
          {'Delete'}
        </Button>
        <DeleteTask
          taskRef={taskRef}
          showDelete={showDelete}
          closeTaskDetails={this.closeTaskDetails}
          closeDeleteModal={this.closeDeleteModal}
        />
      </Modal>
    );
  }
}
