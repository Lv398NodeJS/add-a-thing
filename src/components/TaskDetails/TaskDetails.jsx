import React from 'react';
import { Container, ButtonGroup, Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import EditName from './EditName';
import EditDescription from './EditDescription';
import TaskStatus from './TaskStatus';
import TaskPriority from './TaskPriority';
import * as taskActions from '../../actions/taskDetailsActions';
import './TaskDetailsStyle.scss';
import '../assets/crossicon.svg';
import del from '../assets/delete.svg';

export class TaskDetails extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      editName: false,
      editDescription: false,
    };
  }

  componentDidMount() {
    const { taskRef, taskDetailsActions: { fetchTaskDetails } } = this.props;
    if (!taskRef) return;
    fetchTaskDetails(taskRef);
  }

  handleEditName = () => {
    const { editName } = this.state;
    this.setState({
      editName: !editName,
    });
  };

  handleEditDescription = () => {
    const { editDescription } = this.state;
    this.setState({
      editDescription: !editDescription,
    });
  };

    handleSaveTaskDetails = (updatedName, updatedDescription, updatedStatus, updatedPriority) => {
      const { taskRef, taskDetailsActions: { updateTaskDetails }, taskDetails } = this.props;
      const updatedTaskDetails = {
        name: updatedName || taskDetails.name || {},
        description: updatedDescription || taskDetails.description,
        status: updatedStatus || taskDetails.status,
        priority: updatedPriority || taskDetails.priority,
        subtaskList: taskDetails.subtaskList || {},
      };
      updateTaskDetails(updatedTaskDetails);
      taskRef.set(updatedTaskDetails);
    };

    render() {
      const { editName, editDescription } = this.state;
      const {
        closeTaskDetails, taskDetails = {},
      } = this.props;

      const nameContext = editName
        ? (
          <EditName
            name={taskDetails.name}
            editName={editName}
            closeEditNameField={this.handleEditName}
            saveName={this.handleSaveTaskDetails}
          />
        ) : (
          <Container
            className="open-edit-name"
            onClick={this.handleEditName}
          >
            {taskDetails.name || 'Name'}
          </Container>
        );
      const descriptionContext = editDescription
        ? (
          <EditDescription
            description={taskDetails.description}
            editDescription={editDescription}
            closeEditDescriptionField={this.handleEditDescription}
            saveDescription={this.handleSaveTaskDetails}
          />
        ) : (
          <Container
            className="open-edit-description"
            onClick={this.handleEditDescription}
          >
            {taskDetails.description || 'Some text'}
          </Container>
        );

      return (
        <Container className="task-details pl-0">
          {'Name:'}
          <Button
            className="close-task-button cancel-button float-right px-0"
            onClick={closeTaskDetails}
          >
            <img src={del} alt="close" className="delete-icon" />
          </Button>
          <Container className="task-details-container main-modal-for-taskN">
            {nameContext}
          </Container>
          {'Description:'}
          <Container className="task-details-container description-container main-modal-for-taskD">
            {descriptionContext}
          </Container>
          <ButtonGroup>
            <TaskStatus
              status={taskDetails.status}
              changeStatus={this.handleSaveTaskDetails}
            />
            <TaskPriority
              priority={taskDetails.priority}
              changePriority={this.handleSaveTaskDetails}
            />
          </ButtonGroup>
        </Container>
      );
    }
}

const mapStateToProps = ({ taskDetailsReducer: { taskDetails } }) => ({ taskDetails });

const mapDispatchToProps = dispatch => ({
  taskDetailsActions: bindActionCreators(taskActions, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TaskDetails);
