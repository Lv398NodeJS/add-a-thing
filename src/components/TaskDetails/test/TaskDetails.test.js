import React from 'react';
import { shallow } from 'enzyme';
import TaskDetails from '../TaskDetails';

describe('TaskDetails', () => {
  it('should render correctly', () => {
    const component = shallow(<TaskDetails taskRef={{}} />);

    expect(component).toMatchSnapshot();
  });
});
