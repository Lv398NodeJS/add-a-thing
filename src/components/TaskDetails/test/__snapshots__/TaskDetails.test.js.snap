// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetails should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": Array [
        <ForwardRef
          as="div"
          className="task-details-container"
          fluid={false}
        >
          <ForwardRef
            as="div"
            fluid={false}
            onClick={[Function]}
          />
        </ForwardRef>,
        <ForwardRef
          as="div"
          className="task-details-container description-container"
          fluid={false}
        >
          <ForwardRef
            as="div"
            fluid={false}
            onClick={[Function]}
          >
            Description
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef
          as="div"
          role="group"
          toggle={false}
          vertical={false}
        >
          <TaskStatus
            changeStatus={[Function]}
          />
          <TaskPriority
            changePriority={[Function]}
          />
        </ForwardRef>,
      ],
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef
            as="div"
            fluid={false}
            onClick={[Function]}
          />,
          "className": "task-details-container",
          "fluid": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": undefined,
            "fluid": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "fluid": false,
            },
            "displayName": "Container",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
            "fluid": false,
          },
          "displayName": "Container",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef
            as="div"
            fluid={false}
            onClick={[Function]}
          >
            Description
          </ForwardRef>,
          "className": "task-details-container description-container",
          "fluid": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": "Description",
            "fluid": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Description",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "fluid": false,
            },
            "displayName": "Container",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
            "fluid": false,
          },
          "displayName": "Container",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": Array [
            <TaskStatus
              changeStatus={[Function]}
            />,
            <TaskPriority
              changePriority={[Function]}
            />,
          ],
          "role": "group",
          "toggle": false,
          "vertical": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "changeStatus": [Function],
              "status": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "changePriority": [Function],
              "priority": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
            "role": "group",
            "toggle": false,
            "vertical": false,
          },
          "displayName": "ButtonGroup",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": Array [
          <ForwardRef
            as="div"
            className="task-details-container"
            fluid={false}
          >
            <ForwardRef
              as="div"
              fluid={false}
              onClick={[Function]}
            />
          </ForwardRef>,
          <ForwardRef
            as="div"
            className="task-details-container description-container"
            fluid={false}
          >
            <ForwardRef
              as="div"
              fluid={false}
              onClick={[Function]}
            >
              Description
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef
            as="div"
            role="group"
            toggle={false}
            vertical={false}
          >
            <TaskStatus
              changeStatus={[Function]}
            />
            <TaskPriority
              changePriority={[Function]}
            />
          </ForwardRef>,
        ],
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef
              as="div"
              fluid={false}
              onClick={[Function]}
            />,
            "className": "task-details-container",
            "fluid": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": undefined,
              "fluid": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "fluid": false,
              },
              "displayName": "Container",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "fluid": false,
            },
            "displayName": "Container",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef
              as="div"
              fluid={false}
              onClick={[Function]}
            >
              Description
            </ForwardRef>,
            "className": "task-details-container description-container",
            "fluid": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": "Description",
              "fluid": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Description",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "fluid": false,
              },
              "displayName": "Container",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "fluid": false,
            },
            "displayName": "Container",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <TaskStatus
                changeStatus={[Function]}
              />,
              <TaskPriority
                changePriority={[Function]}
              />,
            ],
            "role": "group",
            "toggle": false,
            "vertical": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changeStatus": [Function],
                "status": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changePriority": [Function],
                "priority": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "role": "group",
              "toggle": false,
              "vertical": false,
            },
            "displayName": "ButtonGroup",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
